### ISSUE TITLE
Consider Checking RFC 6749 Constraints in OAuth2Error

### ISSUE THREAD

**Issue by jzheaux** [spring-projects/spring-security#6831]
`BearerTokenError` validates its constructor parameters against [the character set specified by RFC 6750](https://tools.ietf.org/html/rfc6750#section-3.1). For example:

```java
Assert.isTrue(isScopeValid(scope),
		"scope contains invalid ASCII characters, it must conform to RFC 6750");
```

This is because`BearerTokenError` is a class that intentionally represents OAuth 2.0 bearer token errors from RFC 6750.

We ought to consider having `OAuth2Error` validate its constructor parameters against [the character set specified by RFC 6749](https://tools.ietf.org/html/rfc6749#section-5.2).

`OAuth2Error` appears to be a class that intentionally represents OAuth 2.0 errors from RFC 6749. 

Not verifying that `errorCode`, `description`, and `errorUri` meet this criteria may be a bug. Reports in other libraries demonstrate where [allowing invalid characters can cause problems](https://github.com/spring-projects/spring-security-oauth/issues/889).

**Comment by barclay-reg** [comment]
seems like this is fixing https://github.com/spring-projects/spring-security-oauth/issues/997 and https://github.com/spring-projects/spring-security-oauth/issues/889 finaly, correct?

**Comment by jzheaux** [comment]
@barclay-reg thanks for checking in, however, those tickets are to a separate plugin project. This ticket is referring to Spring Security's built-in support for OAuth 2.0.

**Comment by jgrandja** [comment]
@jzheaux I don't think we should validate `OAuth2Error.description`. See [comment](https://github.com/spring-projects/spring-security-oauth/commit/0367becab8471295952517a40180a55a37679cb8#commitcomment-37474378)

**Comment by rwinch** [comment]
It seems that it might make more sense when we render vs in the Exception itself. Checking in the Exception is a bit like HTML escaping in an Exception just in case it is rendered as an HTML response.
